### PR TITLE
fix(flat-table): correct cursor when using draggable rows

### PR DESCRIPTION
**Pull request by samtjo**
### Proposed behaviour

Change cursor to be `grabbing` when dragging any row anywhere within a table.

Also convert the `flat-table-body-draggable` unit tests to use React Testing Library, which works much better than Enzyme did for this draggable stuff.

### Current behaviour

Currently the `grabbing` cursor disappears when dragging a row around a table

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [x] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

http://localhost:9001/?path=/story/flat-table--with-draggable-rows

### CODE REVIEW

File: src/components/flat-table/flat-table-body-draggable/flat-table-body-draggable.spec.js [hunk 1]
```diff
@@ -1,66 +1,127 @@
...
+      fireEvent.dragEnter(elementToDrag);
+      fireEvent.dragOver(elementToDrag);
+      fireEvent.drop(elementToDrag);
+
+      expect(
+        screen.getAllByRole("row").map((cell) => cell.textContent)
+      ).toEqual(["Row one", "Row two", "Row three"]);
+    });
+
+    test("it can drag-and-drop downwards", () => {
```
Comment (edleeks87) [inline src/components/flat-table/flat-table-body-draggable/flat-table-body-draggable.spec.js:108]: upwards?

File: src/components/flat-table/flat-table-body-draggable/flat-table-body-draggable.spec.js [hunk 2]
```diff
@@ -70,278 +131,254 @@ describe("Draggable Table", () => {
           <FlatTableCell>Edinburgh</FlatTableCell>
```
Comment (edleeks87) [inline src/components/flat-table/flat-table-body-draggable/flat-table-body-draggable.spec.js:131]: Maybe make this Sub Row One if your updating the row text anyway?

File: src/components/flat-table/flat-table-body-draggable/flat-table-body-draggable.spec.js [hunk 3]
```diff
@@ -70,278 +131,254 @@ describe("Draggable Table", () => {
...
-          );
+      expect(
+        screen.getAllByRole("row").map((cell) => cell.textContent)
+      ).toEqual(["Row one", "Row two", "Row three"]);
+    });
 
-          startingNode1.dispatchEvent(
-            createBubbledEvent("dragover", { clientX: 0, clientY: 0 })
-          );
+    test("it can drag-and-drop downwards", () => {
```
Comment (edleeks87) [inline src/components/flat-table/flat-table-body-draggable/flat-table-body-draggable.spec.js:198]: upwards

File: src/components/flat-table/flat-table-body-draggable/flat-table-body-draggable.spec.js (hunk 1 above)
Comment (samtjo) [reply src/components/flat-table/flat-table-body-draggable/flat-table-body-draggable.spec.js:108]: Ah yes, oops 😅

**Review by edleeks87 — APPROVED** [review]

**Review by grabkowski — APPROVED** [review]

**Comment by harpalsingh** [comment]
Need to review if we need another cursor change when hovering over the drag icon but look at that separately.

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**